linode/firewall/framework_models.go

```diff
@@ -0,0 +1,168 @@
+package firewall
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/linode/linodego"
+)
+
+// FirewallModel describes the Terraform resource data model to match the
+// resource schema.
+type FirewallModel struct {
+	ID             types.Int64  `tfsdk:"id"`
+	Label          types.String `tfsdk:"label"`
+	Tags           types.Set    `tfsdk:"tags"`
+	Disabled       types.Bool   `tfsdk:"disabled"`
+	Inbound        types.List   `tfsdk:"inbound"`
+	InboundPolicy  types.String `tfsdk:"inbound_policy"`
+	Outbound       types.List   `tfsdk:"outbound"`
+	OutboundPolicy types.String `tfsdk:"outbound_policy"`
+	Linodes        types.Set    `tfsdk:"linodes"`
+	Devices        types.List   `tfsdk:"devices"`
+	Status         types.String `tfsdk:"status"`
+}
+
+func (data *FirewallModel) parseComputedAttributes(
+	ctx context.Context,
+	firewall *linodego.Firewall,
+	rules *linodego.FirewallRuleSet,
+	devices []linodego.FirewallDevice,
+) diag.Diagnostics {
+	data.ID = types.Int64Value(int64(firewall.ID))
+	data.Disabled = types.BoolValue(firewall.Status == linodego.FirewallDisabled)
```

Comment:
If this function will be shared with resource in the future, I think `disabled` is also non-computed, as defined in the original resource schema.

Reply:
I put it here because it's computed in the datasource schema. https://github.com/linode/terraform-provider-linode/blob/dev/linode/firewall/schema_datasource.go#L67 So when we decide whether we should put it in computed parsing or not, do we only consider if it's computed in the resource schema?

Reply:
> we only consider if it's computed in the resource schema

I think this would be the way.